### DISCUSSION TITLE
사전 식물 삭제 과정에 대해

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#327] [질문있어요] [answered]
어드민 페이지를 개발하면서 사전 식물을 삭제하는 기능을 개발하고있습니다.

이 때 해당 사전 식물을 참조하고있는 반려 식물이 존재하는 경우 어떤 흐름으로 진행해야할까요?

1. 해당 사전 식물을 참조하고있는 반려 식물 모두 삭제
2. 해당 사전 식물을 참조하고있는 반려 식물이 존재할 경우 삭제할 수 없도록 검증 추가

서비스 운영 측면에서는 2번이 옳다고 생각하지만 시스템의 관점에서 본다면 1번도 고려해볼만 하다고 생각됩니다.

사전 식물 삭제 시 어떤 조치를 취해야할까요?

**Comment by WaiNaat** [comment, marked as answer]
저는 2번이 맞다고 생각해요. 이미 식물이 있다면 해당 사전 정보를 지우지 못하게 하거나 다른 사전 식물로 옮긴다음에 지울 수 있게 하는 게 좋아 보입니다

**Comment by yeonkkk** [comment]
저도 2번이 좋습니다. 
사용자 입장에서는 갑자기 반려 식물이 사라진다면 황당할 것 같아요.
참새 말대로 다른 식물로 대체하고 제거하거나, 삭제는 하지 않고 더 이상 해당 사전 식물을 사용할 수 없게 처리해도 좋을 것 같아요(기존에 등록한 반려 식물은 사용 가능하게 하고).

**Comment by Kim0914** [comment]
저도 2번이 좋습니다 ㅎㅎ

사용자가 키우고 있는 반려 식물이 갑자기 삭제된다면 ???????? 물음표 물음표 물음표 할 것 같아요 !

**Comment by bassyu** [comment]
저도 2번에 찬성합니다~!
